### PR TITLE
raftstore: speedup transfer leader

### PR DESCRIPTION
**Pull request by BusyJay**
When doing transfer leader, it will need to wait for at most a round of heartbeat to send out the request, during which the old leader will drop all the requests.

@siddontang @hhkbp2 PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
CI failed

**Comment by siddontang** [comment]
PTAL @hhkbp2

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -90,6 +90,7 @@ pub struct Store<T, C: 'static> {
     pending_raft_groups: HashSet<u64>,
     // region end key -> region id
     region_ranges: BTreeMap<Key, u64>,
+    // the pending snapshots between two mio ticks.
     pending_regions: Vec<metapb::Region>,
```
Comment (hhkbp2) [inline src/raftstore/store/store.rs]: s/pending_regions/pending_snapshot_regions/

File: src/raftstore/store/store.rs [hunk 2]
```diff
@@ -90,6 +90,7 @@ pub struct Store<T, C: 'static> {
     pending_raft_groups: HashSet<u64>,
     // region end key -> region id
     region_ranges: BTreeMap<Key, u64>,
+    // the pending snapshots between two mio ticks.
```
Comment (hhkbp2) [inline src/raftstore/store/store.rs]: s/the pending snapshots/the regions with pending snapshots/

File: src/raftstore/store/peer.rs [hunk 3]
```diff
@@ -943,7 +943,8 @@ impl Peer {
             Ok(RequestPolicy::ProposeNormal) => self.propose_normal(req, metrics),
             Ok(RequestPolicy::ProposeTransferLeader) => {
                 self.propose_transfer_leader(req, cb, metrics);
-                return false;
+                // return true to let raftstore send out the message immediately.
```
Comment (hhkbp2) [inline src/raftstore/store/peer.rs]: change to
// Make the Raft messages to be sent quickly if the transfer leader request is accepted.

**Comment by hhkbp2** [comment]
CI fails on the modified test case.

**Comment by siddontang** [comment]
PTAL @hhkbp2

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**